src/clib/pio_nc.c

```diff
@@ -2452,33 +2546,82 @@ int PIOc_def_var(int ncid, const char *name, nc_type xtype, int ndims,
         file->adios_vars[file->num_vars].nattrs = 0;
         file->adios_vars[file->num_vars].ndims = ndims;
         file->adios_vars[file->num_vars].adios_varid = 0;
+        file->adios_vars[file->num_vars].decomp_varid = 0;
+        file->adios_vars[file->num_vars].frame_varid = 0;
+        file->adios_vars[file->num_vars].fillval_varid = 0;
         file->adios_vars[file->num_vars].gdimids = (int*) malloc(ndims * sizeof(int));
```

Comment:
Can you also add a check for malloc() failure here?

Reply:
OK, done.